### PR TITLE
Add harddrive-iso-single test and "harddrive" tests updates

### PR DESCRIPTION
**Pull request by rvykydal**

### CODE REVIEW

**Comment by rvykydal** [comment]
@jkonecny12 Could you please take a look at the PR ?

**Review by jkonecny12 — APPROVED** [review]
Looks good to me.

File: pre-lib-harddrive.sh [hunk 1]
```diff
@@ -59,3 +55,22 @@ function prepare_tree() {
...
+    mkfs.ext4 -F "${disk}"
+}
+
+
+# Format single partition on a disk for hdd payload source
+function format_single_partition () {
+    local disk="$1"
+
+    sgdisk --zap-all ${disk}
+    sgdisk --new=0:0:+12GiB ${disk}
```
Comment (jstodola) [inline pre-lib-harddrive.sh:74]: Shouldn't the partition size be also increased to 22GiB?

Comment (rvykydal) [reply pre-lib-harddrive.sh:74]: Good point, let me check, we should probably parametrize the function.

Comment (rvykydal) [reply pre-lib-harddrive.sh]: Oh, I guess we need the unallocated space for autopart.

**Comment by rvykydal** [comment]
I removed disabling on fedora as the upstream patch https://github.com/rhinstaller/anaconda/pull/5687 is already merged (daily-iso should pass, rawhide still fail)

**Comment by rvykydal** [comment]
> I am removing disabling on fedora as the upstream patch [rhinstaller/anaconda#5687](https://github.com/rhinstaller/anaconda/pull/5687) is already merged (daily-iso should pass, rawhide still fail)

Indeed: https://github.com/rhinstaller/kickstart-tests/actions/runs/9695026372

**Review by jstodola — APPROVED** [review]
It looks good to me.

**Pull request merged**